### PR TITLE
[melcloud] Adjust to new API limits

### PR DESCRIPTION
**Pull request by thewiep**
Adjust polling values according to new limits introduced by Mitsubishi Electric.
Fixes: #16385

Signed-off-by: Wietse van Buitenen <thewiep@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml [hunk 1]
```diff
@@ -34,10 +34,10 @@
 				<label>Building ID</label>
 				<description>Building ID of the Heatpump device.</description>
 			</parameter>
-			<parameter name="pollingInterval" type="integer" required="true">
+			<parameter name="pollingInterval" type="integer" min="300" required="true">
 				<label>Polling Interval</label>
 				<description>Time interval how often poll data from MELCloud</description>
-				<default>60</default>
+				<default>360</default>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml:40]: Can you also update the readme with this new default?

File: bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml [hunk 2]
```diff
@@ -34,10 +34,10 @@
 				<label>Building ID</label>
 				<description>Building ID of the Heatpump device.</description>
 			</parameter>
-			<parameter name="pollingInterval" type="integer" required="true">
+			<parameter name="pollingInterval" type="integer" min="300" required="true">
```
Comment (lsiepel) [inline bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml:37]: Is there any documentation that supports setting this as a minimum? It seems rather high as a minimum.

File: bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudAccountHandler.java [hunk 3]
```diff
@@ -207,7 +207,7 @@ private void startConnectionCheck() {
                     }
                 }
             };
-            connectionCheckTask = scheduler.scheduleWithFixedDelay(runnable, 0, 60, TimeUnit.SECONDS);
+            connectionCheckTask = scheduler.scheduleWithFixedDelay(runnable, 0, 300, TimeUnit.SECONDS);
```
Comment (lsiepel) [inline bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudAccountHandler.java:210]: Is this change needed? As checking if it is connected or not is not actually causing an API-call, so i guess it should not be affected by the new API limits.

**Review by lsiepel — CHANGES_REQUESTED** [review]
Question and a minor comment.

File: bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml (hunk 1 above)
Comment (thewiep) [reply bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml:40]: Ah yes... missed that one. Thanks, will update.

File: bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml (hunk 2 above)
Comment (thewiep) [reply bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml:37]: As far as I know there isn't any official documentation on this.
Mitsubishi Electric announced limitations on X but didn't specify them:
https://twitter.com/ME_Europe/status/1755205001645801507?s=20
User reports on forum and issue indicate 360 works for them and most, including myself, also have success with 300.
To me it made sense to only allow values that we know will not trigger the "excessive traffic" message from the API.
I can lower it or remove it again if you want though..

File: bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudAccountHandler.java (hunk 3 above)
Comment (thewiep) [reply bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudAccountHandler.java:210]: I changed it as a safety measure as the check itself doesn't perform an API call when the connection is still open but when there's an issue logging in it will trigger an API call every X seconds.
This value being originally 60 will cause the "excessive traffic" message on top of the login issue.
Once they mark your IP as generating excessive traffic, you need to stop connecting for a few hours to reset.
Changing to 300 should prevent that.

File: bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.melcloud/src/main/resources/OH-INF/thing/heatpumpDevice.xml:37]: I would suggest to have it less strict. If users change this to a custom setting, and stumble accross the exception will know how to fix it. Creating a high limit will prevent any user from aadjust it if posslbe

**Comment by thewiep** [comment]
I missed the Signed-off-by line in the second commit, will fix..

**Review by lsiepel — APPROVED** [review]
LGTM, @paulianttila any additional comments?
